src/app/components/core/developer/additional-services/list-additional-services.component.ts

```diff
@@ -0,0 +1,430 @@
+import { Component, OnInit, Input, Output, ViewChild, EventEmitter, OnDestroy, QueryList, ViewChildren } from '@angular/core';
+import { forkJoin } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+
+import { AlertService, ProgressBarService, RolesService, ServicesApiService, SchedulesService, ResponseHandler } from '../../../../services';
+import { SharedService } from '../../../../services/shared.service';
+import { DialogService } from '../../../common/confirmation-dialog/dialog.service';
+import { AdditionalServiceModalComponent } from './additional-service-modal/additional-service-modal.component';
+import { AdditionalServicesContextMenuComponent } from './additional-services-context-menu/additional-services-context-menu.component';
+import { AvailableServices, Schedule, Service } from '../../../../models';
+
+@Component({
+  selector: "app-list-additional-services",
+  templateUrl: "./list-additional-services.component.html",
+  styleUrls: ["./list-additional-services.component.css"],
+})
+export class ListAdditionalServicesComponent implements OnInit, OnDestroy {
+  expectedServices = [
```

Comment:
We could create a helper class for this component and move some of the code there just to make it more readable for example: `expectedServices` array, `applyClass` method etc. Also few lines are getting repeated in `deleteService`, `enableService`, `disableService` methods which could be shifted to one function.

(All the above work could be done in a patch PR)